### PR TITLE
Record the date an apprenticeship is confirmed

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs [hunk 1]
```diff
@@ -63,6 +64,7 @@ public void ConfirmHowApprenticeshipWillBeDelivered(bool howApprenticeshipDelive
         public void ConfirmApprenticeship(bool apprenticeshipCorrect)
         {
             ApprenticeshipConfirmed = apprenticeshipCorrect;
+            ConfirmedOn = DateTime.Now.Date;
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs:67]: Do you need to check that all sections have been confirmed? Is that a domain requirement? 

Also can an apprentice Unconfirm a CS? If not do they need the param? And teh ApprenticeshipConfirmed field could then be calculated

**Review by Pauljgraham — COMMENTED** [review]
One small question

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs (hunk 1 above)
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs:67]: Yes, that's probably a good idea to confirm the confirmation of all sections before allowing the overall confirmation.

I don't believe an apprentice can unconfirm (I haven't seen a requirement to that effect).

However, just having all the individual sections confirmed is not the same as confirming the apprenticeship itself.  That needs to be an explicit action by the apprentice.

I suppose we could treat the presence of `ConfirmedOn` to be the indication that the apprenticeship is confirmed - which is probably what you were suggesting?  I quite like that idea as it prevents the possibility that the database has `ApprenticeshipConfirmed = true` and `ConfirmedOnDate = null` or vice-versa.

Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs]: I have applied both suggestions

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs [hunk 2]
```diff
@@ -63,8 +64,20 @@ public void ConfirmHowApprenticeshipWillBeDelivered(bool howApprenticeshipDelive
...
         {
-            ApprenticeshipConfirmed = apprenticeshipCorrect;
-            ConfirmedOn = DateTime.Now.Date;
+            if (apprenticeshipCorrect
+                && TrainingProviderCorrect == true
+                && EmployerCorrect == true
+                && RolesAndResponsibilitiesCorrect == true
+                && ApprenticeshipDetailsCorrect == true
+                && HowApprenticeshipDeliveredCorrect == true
+                && HowApprenticeshipDeliveredCorrect == true)
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs:73]: duplicate line

**Review by Pauljgraham — APPROVED** [review]
Approved just a duplicate line

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs (hunk 2 above)
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Data/Models/CommitmentStatement.cs:72]: Duplicate removed.

**Pull request merged**